src/main/java/com/cburch/logisim/fpga/hdlgenerator/LedArrayColumnScanningHDLGeneratorFactory.java

```diff
@@ -190,119 +191,120 @@
+          "",
+          "s_scanningCounterNext <= (OTHERS => '0') WHEN s_tickReg /= '0' AND s_tickReg /= '1' ELSE -- for simulation",
+          "                         std_logic_vector(to_unsigned({{counterValue}}-1, {{counterBits}}))",
+          "                            WHEN s_scanningCounterReg = std_logic_vector(to_unsigned(0, {{counterBits}})) ELSE ",
+          "                         std_logic_vector(unsigned(s_scanningCounterReg)-1);",
+          "",
+          "s_columnCounterNext <= (OTHERS => '0') WHEN s_tickReg /= '0' AND s_tickReg /= '1' ELSE -- for simulation",
+          "                       s_columnCounterReg WHEN s_tickReg = '0' ELSE",
+          "                       std_logic_vector(to_unsigned(nrOfColumns-1,nrOfcolumnAddressBits))",
+          "                          WHEN s_columnCounterReg = std_logic_vector(to_unsigned(0,nrOfColumnAddressBits)) ELSE",
+          "                       std_logic_vector(unsigned(s_columnCounterReg)-1);",
+          "",
+          "makeFlops : PROCESS ({{clock}}) IS",
+          "BEGIN",
+          "   IF (rising_edge({{clock}})) THEN",
+          "      s_columnCounterReg   <= s_columnCounterNext;",
+          "      s_scanningCounterReg <= s_scanningCounterNext;",
+          "      s_tickReg            <= s_tickNext;",
+          "   END IF;",
+          "END PROCESS makeFlops;",
+          "");
     } else {
-      contents.add("");
-      contents.add("   assign columnAddress = s_columnCounterReg;");
-      contents.add("");
-      contents.add("   assign s_tickNext = (s_scanningCounterReg == 0) ? 1'b1 : 1'b0;");
-      contents.add("   assign s_scanningCounterNext = (s_scanningCounterReg == 0) ? "
-              + scanningCounterValueString 
-              + " : s_scanningCounterReg - 1;");
-      contents.add("   assign s_columnCounterNext = (s_tickReg == 1'b0) ? s_columnCounterReg : ");
-      contents.add("                                (s_rowCounterReg == 0) ? nrOfColumns-1 : s_columnCounterReg-1;");
-      contents.add("");
-      contents.addAll(MakeRemarkBlock("Here the simulation only initial is defined", 3));
-      contents.add("   initial");
-      contents.add("   begin");
-      contents.add("      s_columnCounterReg   = 0;");
-      contents.add("      s_scanningCounterReg = 0;");
-      contents.add("      s_tickReg            = 1'b0;");
-      contents.add("   end");
-      contents.add("");
-      contents.add("   always @(posedge " + TickComponentHDLGeneratorFactory.FPGAClock + ")");
-      contents.add("   begin");
-      contents.add("       s_columnCounterReg   = s_columnCounterNext;");
-      contents.add("       s_scanningCounterReg = s_scanningCounterNext;");
-      contents.add("       s_tickReg            = s_tickNext;");
-      contents.add("   end");
-
+      contents
+          .add(
+              "",
+              "assign columnAddress = s_columnCounterReg;",
```

Comment:
should columnAddress not be {{columnAddress}} to be consistent with the VHDL counterpart?

Reply:
Even if so, I'd not do that as part of this PR. I tried not to alter HDL if possible, because that's not in scope here. In future, as editing should be now simpler it may be a good idea to review all the HDLs and simplify for for how, this is how it was up to now.

Reply:
It will not alter the HDL, as in the VHDL part you replaced columnAddress by  {{columnAddress}}, hence I would expect in the Verilog part the same "generic", but I agree with you that we should do a second pass.